front50-azure/src/main/java/com/netflix/spinnaker/front50/model/AzureStorageService.java

```diff
@@ -0,0 +1,319 @@
+        blobContainer.createIfNotExists();
+        BlobContainerPermissions permissions = new BlobContainerPermissions();
+        permissions.setPublicAccess(BlobContainerPublicAccessType.CONTAINER);
+        blobContainer.uploadPermissions(permissions);
+      }
+      catch (Exception e) {
+        // log exception
+        blobContainer = null;
+      }
+    }
+    return blobContainer;
+  }
+
+  public AzureStorageService(String connectionString, String storageAccountName, String containerName, String rootFolder) {
+    this.storageAccountName = storageAccountName;
+    this.containerName = containerName;
+    this.rootFolder = rootFolder;
+    try {
+      this.storageAccount = CloudStorageAccount.parse(connectionString);
+    } catch (Exception e) {
+      // Log the exception
+      this.storageAccount = null;
+    }
+  }
+
+  @Override
+  public void ensureBucketExists() {
+    // do nothing
+  }
+
+  @Override
+  public boolean supportsVersioning() {
+    return true;
+  }
+
+  @Override
+  public <T extends Timestamped> T loadObject(ObjectType objectType, String objectKey) {
+    String key = buildKeyPath(objectType.group, objectKey, objectType.defaultMetadataFilename);
+    try {
+      CloudBlockBlob blob = getBlobContainer().getBlockBlobReference(key);
+      if (blob.exists()) {
+        return deserialize(blob, (Class<T>) objectType.clazz);
+      }
+    } catch (StorageException se) {
+      logStorageException(se, key);
+    }
+    catch (Exception e) {
+      log.error("Failed to retrieve {} object: {}: {}", objectType.group, key, e.getMessage());
+    }
+    return null;
+  }
+
+  @Override
+  public <T extends Timestamped> Collection<T> loadObjectsWithPrefix(ObjectType objectType, String prefix, int maxResults) {
+    Set<T> blobs = new HashSet<>();
+    String key = buildKeyPath(objectType.group, prefix, "");
+    try {
+      ResultContinuation token = null;
+      EnumSet<BlobListingDetails> listDetails = EnumSet.of(BlobListingDetails.METADATA);
+      do {
+        ResultSegment<ListBlobItem> result = blobContainer.listBlobsSegmented(key, true, listDetails, maxResults, token, null, null);
+        token = result.getContinuationToken();
+
+        for (ListBlobItem item : result.getResults()) {
+          blobs.add(deserialize((CloudBlockBlob)item, (Class<T>) objectType.clazz));
+        }
+      } while (token != null);
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to deserialize object(s) (key: " + key + ")", e);
+    } catch (StorageException se) {
+      logStorageException(se, key);
+    } catch (Exception e) {
+      log.error("Failed to retrieve objects with prefix {}: {}", key, e.getMessage());
+    }
+    return blobs;
+  }
+
+  @Override
+  public <T extends Timestamped> T loadObjectVersion(ObjectType objectType, String objectKey, String versionId) throws NotFoundException {
+    throw new UnsupportedOperationException("loadObjectVersion() is not supported");
```

Comment:
Sorry, no objection to how it is now. I'll dive a bit more into why this method isn't implemented anywhere next week. It's definitely curious.